### ISSUE TITLE
Is it available to get last time step with return_sequences=True?

### ISSUE THREAD

**Issue by poyuwu** [fchollet/keras#2951]
As title, is it available to access last time step when recurrent layer's option,'return_sequences=True'? 

(As following picture, since deep recurrent need each time step, 'g' need only last time step)
ref: [http://arxiv.org/pdf/1506.03340v3.pdf](http://arxiv.org/pdf/1506.03340v3.pdf)
![screenshot from 2016-06-12 01 02 21](https://cloud.githubusercontent.com/assets/1275548/15986396/92b5b718-3039-11e6-8a84-900e65a728f8.png)

Besides,
can keras merge 2D shape(ex: (None,300)) and 3D shape(ex: (None,10,300)) do column-wise cosine similarity? That is, I'll get output_shape = (None,10)

**Comment by braingineer** [comment]
you just use a lambda layer:

```python
Lambda(lambda x: return x[:,-1,:])
```

**Comment by poyuwu** [comment]
1.  
<p> a) I tried to use Lambda layer, but I have output_shape error, that is I have the same output_shape as first layer (though ndim is 2)
<p> b) If I just use `first_layer_recurrent[:,-1,:]` to next layer input, then I have Exception error as following 
```Exception: You tried to call layer "dropout_10". This layer has no information about its expected input shape, and thus cannot be built. You can build it manually via: `layer.build(batch_input_shape)` ```
2. 
I tried to use mode='dot' to get cosine similarity because mode='cos' used in 3D shape is not better than I expected. (which means its output_shape is (None,1) ).
